src/pages/task/actionButtons/PreviousCommits.tsx

```diff
@@ -0,0 +1,198 @@
+            ...bvOptionsBase,
+            statuses: finishedTaskStatuses,
+          },
+        },
+      });
+    }
+  }, [
+    baseTaskId,
+    baseTaskStatus,
+    buildVariant,
+    displayName,
+    fetchLastExecuted,
+    fetchLastPassing,
+    isBaseTaskFinished,
+    lastExecutedCalled,
+    lastPassingCalled,
+    projectIdentifier,
+    selectState,
+    skipOrderNumber,
+  ]);
+
+  const lastPassingTaskId = getTaskIdFromMainlineCommitsQuery(lastPassingData);
+  const lastExecutedTaskId = getTaskIdFromMainlineCommitsQuery(
+    lastExecutedData
+  );
+  let link = "";
+  switch (selectState) {
+    case "base":
+      // The task may not exist on the base version if it was generated.
+      link = baseTaskId ? getTaskRoute(baseTaskId) : "";
+      break;
+    case "lastPassing":
+      // If a base task succeeded, the last passing commit is the base task.
+      if (baseTaskStatus === TaskStatus.Succeeded) {
+        link = getTaskRoute(baseTaskId);
+      } else if (lastPassingTaskId) {
+        link = getTaskRoute(lastPassingTaskId);
+      } else {
+        link = "";
+      }
+      break;
+    case "lastExecuted":
+      // If a base task exists and has finished, the last executed commit is the base task.
+      if (isBaseTaskFinished) {
+        link = getTaskRoute(baseTaskId);
+      } else if (lastExecutedTaskId) {
+        link = getTaskRoute(lastExecutedTaskId);
+      } else {
+        link = "";
+      }
+      break;
+    default:
+  }
+
+  return versionMetadata?.isPatch === true ? (
+    <Container>
+      <StyledSelect
+        size="small"
+        label="Previous commits for this task"
+        onChange={(v) => setSelectState(v as commitType)}
+        data-cy="previous-commits"
+        allowDeselect={false}
+        value={selectState}
+        disabled={!versionMetadata?.baseVersion}
+      >
+        <Option value="base" key="base">
+          Go to base commit
+        </Option>
+        <Option value="lastPassing" key="lastPassing">
+          Go to last passing version
+        </Option>
+        <Option value="lastExecuted" key="lastExecuted">
+          Go to last executed version
+        </Option>
+      </StyledSelect>
+      <Button as={Link} to={link} disabled={!link} size="small">
+        Go
+      </Button>
+    </Container>
+  ) : null;
+};
+
+const Container = styled.div`
+  display: flex;
+`;
+
+// @ts-expect-error
+const StyledSelect = styled(Select)`
+  margin-right: 8px;
+  margin-top: -20px;
+`;
+
+const getTaskIdFromMainlineCommitsQuery = (
+  data: GetLastMainlineCommitQuery
+) => {
+  const buildVariants =
+    data?.mainlineCommits.versions.find(({ version }) => version)?.version
+      .buildVariants ?? [];
+  if (buildVariants.length > 1) {
+    reportError(
+      "Multiple build variants matched previous commit search."
+    ).warning();
+  }
+  if (buildVariants[0]?.tasks.length > 1) {
+    reportError("Multiple tasks matched previous commit search.").warning();
+  }
```

Comment:
I don't think this scenario would occur unless there was an issue with the data model, resolver, or fetch parameter. I thought the reportError is a good sanity check to prevent the component from failing silently since we depend on the list having length 1 which I didn't think was an explicit guarantee